File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* replaceElements(int* arr, int arrSize, int* returnSize) {
int* result = (int*)malloc(arrSize * sizeof(int));
*returnSize = arrSize;

    int maxRight = -1; 
    for (int i = arrSize - 1; i >= 0; i--) {
        result[i] = maxRight;       
        if (arr[i] > maxRight) {
            maxRight = arr[i];       
        }
    }

    return result;
}

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int arrSize = arr.length;
        int[] sum = new int[arrSize];

        for (int i = 0; i < arrSize - 1; i++) {
            int max = arr[i + 1];

            for (int j = i + 1; j < arrSize; j++) {
                if (arr[j] > max) {
                    max = arr[j];
                }
            }
            sum[i] = max;
        }

        sum[arrSize - 1] = -1;   

        return sum;
    }
}
